File: ByteDance/101. 对称二叉树.go
package main

// 101. 对称二叉树

// 给定一个二叉树，检查它是否是镜像对称的。

func main() {

}

func isSymmetric(root *TreeNode) bool {
	return isSymme(root, root)
}

func isSymme(root1, root2 *TreeNode) bool {
	if root1 == nil && root2 == nil {
		return true
	}
	if root1 == nil || root2 == nil {
		return false
	}
	if root1.Val != root2.Val {
		return false
	}

	return isSymme(root1.Left, root2.Right) && isSymme(root1.Right, root2.Left)
}

File: C++/101.对称二叉树.cpp

// 101. 对称二叉树
//
// 给定一个二叉树，检查它是否是镜像对称的。
// 例如，二叉树 [1,2,2,3,4,4,3] 是对称的。

#include "header.h"

class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return isSymme(root, root);
    }

    bool isSymme(TreeNode* left, TreeNode* right) {
        if (left == nullptr && right == nullptr) {
            return true;
        }
        if (left == nullptr || right == nullptr) {
            return false;
        }
        if (left->val != right->val) {
            return false;
        }
        return isSymme(left->left, right->right) && isSymme(left->right, right->left);
    }
};
